### PR TITLE
Rename JS<T> to Dom<T>, Root<T> to DomRoot<T>, and other things

### PR DESCRIPTION
**Pull request by nox**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/18635)

### CODE REVIEW

File: components/script/docs/JS-Servos-only-GC.md [hunk 1]
```diff
@@ -164,8 +164,8 @@ relationship. The `Document` just has a pointer to a `Window`, one of many
 pointers to that object, which can live in native DOM data structures or in
 JavaScript objects. These are precisely the pointers we need to tell the
 garbage collector about. We do this with a
-[custom type for traced pointers: `JS<T>`][js] (for example, the `JS<Window>`
-above). The implementation of `trace` for `JS<T>` is not auto-generated; this
+[custom type for traced pointers: `Dom<T>`][js] (for example, the `Dom<Window>`
```
Comment (emilio) [inline components/script/docs/JS-Servos-only-GC.md]: nit: Maybe change the link name too? (right now it's called `js`).

**Review by emilio — APPROVED** [review]
rs=me, given you and @jdm discussed this yesterday, and this looks fine to me.

Kinda unfortunate that we have both `Dom` and `DOM` stuff though...

File: components/script/docs/JS-Servos-only-GC.md (hunk 1 above)
Comment (nox) [reply components/script/docs/JS-Servos-only-GC.md]: Done.

**Pull request merged**